--- /home/jsloan/src/libs3-2.0-org/src/request.c	2011-09-28 02:16:09.000000000 -0600
+++ /home/jsloan/src/libs3-2.0/src/request.c	2011-12-12 11:11:41.218842857 -0700
@@ -28,6 +28,7 @@
 #include <pthread.h>
 #include <stdlib.h>
 #include <string.h>
+#include <unistd.h> /* < HAYLOFT coverclock@diag.com 2011-12-12 */
 #include <sys/utsname.h>
 #include "request.h"
 #include "request_context.h"
@@ -796,6 +797,8 @@ static S3Status setup_curl(Request *requ
                            const RequestComputedValues *values)
 {
     CURLcode status;
+	const char *hayloft_verbose; /* < HAYLOFT coverclock@diag.com 2011-12-12 */
+	const char *hayloft_proxy; /* < HAYLOFT coverclock@diag.com 2011-12-12 */
 
 #define curl_easy_setopt_safe(opt, val)                                 \
     if ((status = curl_easy_setopt                                      \
@@ -806,6 +809,13 @@ static S3Status setup_curl(Request *requ
     // Debugging only
     // curl_easy_setopt_safe(CURLOPT_VERBOSE, 1);
     
+    /* v HAYLOFT coverclock@diag.com 2011-12-12 */
+    hayloft_verbose = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_VERBOSE");
+    if (hayloft_verbose != (char *)0) {
+        curl_easy_setopt_safe(CURLOPT_VERBOSE, atol(hayloft_verbose));
+    }
+    /* ^ HAYLOFT coverclock@diag.com 2011-12-12 */
+    
     // Set private data to request for the benefit of S3RequestContext
     curl_easy_setopt_safe(CURLOPT_PRIVATE, request);
     
@@ -837,6 +847,15 @@ static S3Status setup_curl(Request *requ
     // xxx todo - support setting the proxy for Curl to use (can't use https
     // for proxies though)
 
+    /* v HAYLOFT coverclock@diag.com 2011-12-12 */
+    hayloft_proxy = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_PROXY");
+    /* Nominally this will be "127.0.0.1:8888" if it exists. */
+    if (hayloft_proxy != (char *)0) {
+        curl_easy_setopt_safe(CURLOPT_PROXY, hayloft_proxy);
+        curl_easy_setopt_safe(CURLOPT_HTTPPROXYTUNNEL, 1L);
+    }
+    /* ^ HAYLOFT coverclock@diag.com 2011-12-12 */
+
     // xxx todo - support setting the network interface for Curl to use
 
     // I think this is useful - we don't need interactive performance, we need
