diff -purN /home/jsloan/src/libs3-2.0-orig/src/request.c /home/jsloan/src/libs3-2.0/src/request.c && false || true
--- /home/jsloan/src/libs3-2.0-orig/src/request.c	2017-11-02 11:31:16.298814521 -0600
+++ /home/jsloan/src/libs3-2.0/src/request.c	2017-11-02 11:34:37.969880821 -0600
@@ -805,6 +805,16 @@ static S3Status setup_curl(Request *requ
 
     // Debugging only
     // curl_easy_setopt_safe(CURLOPT_VERBOSE, 1);
+    // v https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
+    {
+        extern char * getenv(const char *);
+        const char * com_diag_hayloft_option;
+        com_diag_hayloft_option = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_VERBOSE");
+        if (com_diag_hayloft_option) {
+            curl_easy_setopt_safe(CURLOPT_VERBOSE, atol(com_diag_hayloft_option));
+        }
+    }
+    // ^ https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
     
     // Set private data to request for the benefit of S3RequestContext
     curl_easy_setopt_safe(CURLOPT_PRIVATE, request);
@@ -836,6 +846,17 @@ static S3Status setup_curl(Request *requ
 
     // xxx todo - support setting the proxy for Curl to use (can't use https
     // for proxies though)
+    // v https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
+    {
+        extern char * getenv(const char *);
+        const char * com_diag_hayloft_option;
+        com_diag_hayloft_option = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_PROXY");
+        if (com_diag_hayloft_option) {
+            curl_easy_setopt_safe(CURLOPT_PROXY, com_diag_hayloft_option);
+            curl_easy_setopt_safe(CURLOPT_HTTPPROXYTUNNEL, 1L);
+        }
+    }
+    // ^ https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
 
     // xxx todo - support setting the network interface for Curl to use
 
@@ -866,6 +887,20 @@ static S3Status setup_curl(Request *requ
     // xxx todo - allow configurable max send and receive speed
     curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, 1024);
     curl_easy_setopt_safe(CURLOPT_LOW_SPEED_TIME, 15);
+    // v https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
+    {
+        extern char * getenv(const char *);
+        const char * com_diag_hayloft_option;
+        com_diag_hayloft_option = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_LOW_SPEED_LIMIT");
+        if (com_diag_hayloft_option) {
+            curl_easy_setopt_safe(CURLOPT_LOW_SPEED_LIMIT, atol(com_diag_hayloft_option));
+        }
+        com_diag_hayloft_option = getenv("COM_DIAG_HAYLOFT_LIBS3_CURL_LOW_SPEED_TIME");
+        if (com_diag_hayloft_option) {
+            curl_easy_setopt_safe(CURLOPT_LOW_SPEED_TIME, atol(com_diag_hayloft_option));
+        }
+    }
+    // ^ https://github.com/coverclock/com-diag-hayloft coverclock@diag.com 2017-11-02
 
     // Append standard headers
 #define append_standard_header(fieldName)                               \
